app/models/user_ranking.rb

```diff
@@ -1,21 +1,5 @@
 class UserRanking
-  class << self
-    def users_array
-      User.find(all.entries.collect(&:_id))
-    rescue Mongoid::Errors::DocumentNotFound
-      []
-    end
-
-    def users_hash
-      Hash[*users_array.collect { |u| [u._id, u] }.flatten]
-    end
-
-    def users
-      @users ||= users_hash
-    end
-  end
-
   def user
-    @user ||= UserRankingAllTime.users[_id]
+    @user ||= User.where(_id: _id).first
   end
```

Comment:
The actual implementation isn't working well. If you check the current production code, something about the user relation is going wrong.

Moreover I don't think that putting all the users collection in memory (into a class variable) on the server startup is a good idea. After that new registered user will never appear into the leaderboards until a server restart.

N+1 queries are covered by the fragment cache on the view.